app/models/foreman_salt/concerns/hostgroup_extensions.rb

```diff
@@ -28,31 +28,29 @@ def inherited_salt_module_ids
       end
 
       def salt_proxy
-        return super unless ancestry.present?
-        SmartProxy.find_by_id(inherited_salt_proxy_id)
-      end
-
-      def inherited_salt_proxy_id
         if ancestry.present?
```

Comment:
Reading this suggests it reverses the order. However, the `inherited_salt_proxy_id` prefers the actual ID which is a bit confusing to me. Perhaps you should use `real_salt_proxy_id` or `actual_salt_proxy_id`.

Reply:
same behavior is in foreman_chef. Therefore I would stick with this for now.